sdk/storage/src/authorization/mod.rs

```diff
@@ -136,52 +131,35 @@ impl StorageCredentials {
     ///
     /// This method is useful for updating credentials that are used by multiple
     /// clients at once.
```

Comment:
By making this take `&mut self`, you cannot update multiple clients at once.

Can you discuss how this would work for those cases?

Reply:
Thank you for the review. Could you exemplify what you mean with an example? I could not fund a test that would no longer work with this PR.

Reply:
@jorgecarleitao , the underlying reason for mutex wasn't exposed appropriately.  

#1474 exposes the intended functionality.

Reply:
@jorgecarleitao , with the change in #1477, the impact performance from using a Mutex should be reduced.